lib/krane/kubernetes_resource/stateful_set.rb

```diff
@@ -22,26 +22,15 @@ def status
     def deploy_succeeded?
       success = observed_generation == current_generation
 
-      @pods.each do |pod|
-        success &= pod.definition["metadata"]["labels"]["controller-revision-hash"] == status_data['updateRevision']
-      end
-
-      if update_strategy == 'RollingUpdate'
-        success &= status_data['currentRevision'] == status_data['updateRevision']
-        success &= desired_replicas == status_data['readyReplicas'].to_i
-        success &= desired_replicas == status_data['currentReplicas'].to_i
-
-      elsif update_strategy == ONDELETE
+      if update_strategy == ONDELETE && required_rollout != "full"
```

Comment:
Are there known cases of `update_strategy == DELETE` AND `required_rollout == "full"`? If those conditions hold, `krane` will time out unless something external deletes the pods, right? I suppose that's not a problem if that's what's been encoded in the manifest since that would signal that as the desired behaviour 🤔

Reply:
I do know of [one such case](https://github.com/Shopify/kafka-infrastructure/blob/main/config/deploy/partials/kafka.yml.erb#L130) in `kafka-infrastructure`, where [this script](https://github.com/Shopify/zonal-deploy-controller/blob/main/pkg/controllers/statefulset_controller.go) in `zonal-deploy-controller` is used to delete the pods. I don't know that this change would break anything there, but tagging @Shopify/streaming-foundations for visibility